generator/src/main/java/org/corfudb/generator/Generator.java

```diff
@@ -19,55 +19,196 @@
  * patterns can be approximated by setting different configurations like
  * data distributions, concurrency level, operation types, time skews etc.
  *
- *
  * Created by maithem on 7/14/17.
+ *
+ * Modified by Miao Cheng on 10/15/18:
+ *
+ * Re-implemented the Generator class to provide APIs for running workloads that can be
+ * 'divided' into chunks, of which transactions are decoupled from each other.
+ * This can help do correctness verification for Longevity test chunk-by-chunk on the fly.
  */
+@Slf4j
 public class Generator {
-    public static void main(String[] args) {
-        String endPoint = args[0];
-        int numStreams = Integer.parseInt(args[1]);
-        int numKeys = Integer.parseInt(args[2]);
-        long cpPeriod = Long.parseLong(args[3]);
-        int numThreads = Integer.parseInt(args[4]);
-        int numOperations = Integer.parseInt(args[5]);
 
-        CorfuRuntime rt = new CorfuRuntime(endPoint).connect();
+    private long cpPeriod;
+    private Duration totalRuntime;
+
+    static final int numStreams = 50;
+    static final int numKeys = 100;
+    static final int numThreads = 10;
+
+    BlockingQueue<Operation> operationQueue;
+    CorfuRuntime rt;
+    State state;
+    ScheduledExecutorService scheduler;
+    ExecutorService workers;
+
+    // The lock prevents consuming the queue while it's being filled.
+    Boolean operationQueueLock;
```

Comment:
This variable should probably be `volatile`. Also you can make it `boolean`instead of `Boolean`
Actually I think we can remove this lock variable since `BlockingQueue` implementations are all thread safe, including `ArrayBlockingQueue`. What do you think?

Reply:
Didn't notice there is a difference between 'boolean' and 'Boolean'... will change it to a boolean. I know BlockingQueue is thread safe, but it's for doing one thing at a time. However, while filling the queue, we are putting many operations one by one into the queue (see the for loop in fillOperationQueue), so I added this lock for more safety. Anyway, we can discuss this more offline.